lib/Dialect/AIE/Transforms/AIEAssignBuffers.cpp

```diff
@@ -66,16 +66,28 @@ LogicalResult basicAllocation(TileOp &tile) {
     maxDataMemorySize = targetModel.getLocalMemorySize();
 
   SmallVector<BufferOp, 4> buffers;
-  // Collect all the buffers for this tile.
+  std::list<BufferOp> allocated_buffers;
+  // Collect all the buffers for this tile. If the buffer has an address, add
+  // it to allocated_buffers. Otherwise, add it to buffers.
   device.walk<WalkOrder::PreOrder>([&](BufferOp buffer) {
-    if (buffer.getTileOp() == tile)
-      buffers.push_back(buffer);
+    if (buffer.getTileOp() == tile) {
+      if (buffer.getAddress())
+        allocated_buffers.push_back(buffer);
+      else
+        buffers.push_back(buffer);
+    }
   });
-  // Sort by allocation size.
+
+  // Sort buffers by allocation size.
   std::sort(buffers.begin(), buffers.end(), [](BufferOp a, BufferOp b) {
     return a.getAllocationSize() > b.getAllocationSize();
   });
 
+  // Sort allocated_buffers by address
+  allocated_buffers.sort([](BufferOp a, BufferOp b) {
```

Comment:
Interesting, I did not remember `std::list` had a `sort()` member!
Any reason for not using a `std::vector`, besides being a Halloween programming exercise? :ghost: :jack_o_lantern: :smile: